Lib/glyphsLib/builder/layers.py

```diff
@@ -26,6 +29,19 @@ def to_ufo_layer(self, glyph, layer):
         ufo_layer = ufo_font.layers.defaultLayer
     elif layer.name not in ufo_font.layers:
         ufo_layer = ufo_font.newLayer(layer.name)
+    elif (layer.name in ufo_font.layers and
+            glyph.name in ufo_font.layers[layer.name]):
+        logger.warning(
+            "Glyph {}, layer {}: Duplicate glyph layer name".format(
+                glyph.name, layer.name
```

Comment:
instead of using `.format` you can pass positional arguments to the logger.warning() call (using `%s` for string interpolation which is the default for logging). They will only be string-interpolated if the message is effectively logged (e.g. there's a handler, it is in level, etc.)